core/src/main/java/com/graphhopper/routing/util/parsers/OSMBarrierParser.java

```diff
@@ -0,0 +1,43 @@
+package com.graphhopper.routing.util.parsers;
+
+import com.graphhopper.reader.ReaderWay;
+import com.graphhopper.routing.ev.Barrier;
+import com.graphhopper.routing.ev.EdgeIntAccess;
+import com.graphhopper.routing.ev.EnumEncodedValue;
+import com.graphhopper.storage.IntsRef;
+import com.graphhopper.util.Helper;
+
+import java.util.List;
+import java.util.Map;
+
+public class OSMBarrierParser implements TagParser {
+    protected final EnumEncodedValue<Barrier> barrierEnc;
+
+    public OSMBarrierParser(EnumEncodedValue<Barrier> barrierEnc) {
+        this.barrierEnc = barrierEnc;
+    }
+
+    @Override
+    public void handleWayTags(int edgeId, EdgeIntAccess edgeIntAccess, ReaderWay way, IntsRef relationFlags) {
+        if (way.hasTag("gh:barrier_edge")) return;
```

Comment:
Shouldn't this be `!way.hasTag(...)`? The ways with the `gh:barrier_edge` are the ones we are interested it, no?